glean-core/src/ping/mod.rs

```diff
@@ -67,12 +70,80 @@ impl PingMaker {
         current_seq as usize
     }
 
+    fn get_start_end_times(&self, glean: &Glean, storage_name: &str) -> (Metric, Metric) {
+        let time_unit = TimeUnit::Minute;
+
+        let start_time = DatetimeMetric::new(
+            CommonMetricData {
+                name: format!("{}#start", storage_name),
+                category: "".into(),
+                send_in_pings: vec![INTERNAL_STORAGE.into()],
+                lifetime: Lifetime::User,
+                ..Default::default()
+            },
+            time_unit,
+        );
+
+        let end_time = DatetimeMetric::new(
+            CommonMetricData {
+                name: format!("{}#end", storage_name),
+                category: "".into(),
+                send_in_pings: vec![INTERNAL_STORAGE.into()],
+                lifetime: Lifetime::User,
+                ..Default::default()
+            },
+            time_unit,
+        );
+
+        let start_time_data = match StorageManager.snapshot_metric(
+            glean.storage(),
+            INTERNAL_STORAGE,
+            &start_time.meta().identifier(),
+        ) {
+            Some(Metric::Datetime(t, tu)) => {
+                log::info!("datetime start found: {:?}", t);
+                Metric::Datetime(t, tu)
+            }
+            _ => {
+                let now: DateTime<Local> = Local::now();
```

Comment:
Can we refactor this into some utils file as `get_local_time_with_offset` (or some better naming)?